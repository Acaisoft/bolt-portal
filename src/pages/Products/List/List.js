import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { Link } from 'react-router-dom'
import { Button, IconButton } from '@material-ui/core'
import { Info, Edit } from '@material-ui/icons'
import DataTable from '~components/DataTable'

import fakeProducts from '../productsData.mock'

export class List extends Component {
  static propTypes = {
    match: PropTypes.shape({
      url: PropTypes.string.isRequired,
    }).isRequired,
  }

  render() {
    const { match } = this.props

    return (
      <div>
        <div>Products list</div>
        <Button
          variant="contained"
          color="primary"
          component={Link}
          to={`${match.url}/create`}
        >
          Create a new product
        </Button>
        <DataTable data={fakeProducts}>
          <DataTable.Column key="id" render={x => x.id} title="ID" />
          <DataTable.Column key="name" render={x => x.name} title="Name" />
          <DataTable.Column key="actions" render={this.renderActions} />
        </DataTable>
      </div>
    )
  }

  renderActions = row => {
    const { match } = this.props
    return (
      <div>
        <IconButton
          component={Link}
          to={`${match.url}/${row.id}/details`}
          aria-label="Show details"
        >
          <Info />
        </IconButton>
        <IconButton
          component={Link}
          to={`${match.url}/${row.id}/edit`}
          aria-label="Edit product"
        >
          <Edit />
        </IconButton>
      </div>
    )
  }
}

export default List
