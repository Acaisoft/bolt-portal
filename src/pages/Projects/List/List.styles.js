export default ({ palette, spacing, typography }) => {
  return {
    root: {
      flexGrow: 1,
    },
    btnContainer: {
      display: 'flex',
      justifyContent: 'flex-end',
      marginBottom: 20,
    },
    card: {
      display: 'flex',
      justifyContent: 'center',
      textAlign: 'center',
    },
  }
}
