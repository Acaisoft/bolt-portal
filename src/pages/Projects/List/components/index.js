export { default as BackgroundImage } from './BackgroundImage'
export { default as NewProjectCard } from './NewProjectCard'
export { default as ProjectCard } from './ProjectCard'
export { default as ProjectFormInCard } from './ProjectFormInCard'
export { default as ProjectsList } from './ProjectsList'
