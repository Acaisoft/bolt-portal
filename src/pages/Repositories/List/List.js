import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { Mutation } from 'react-apollo'
import { Typography, withStyles } from '@material-ui/core'
import { AddButton, DeleteModal } from '~components'
import { TestSourcesList } from '~containers/lists'

import styles from './List.styles'
import { GET_TEST_SOURCES_QUERY } from '~services/GraphQL/Queries'
import { DELETE_REPOSITORY_MUTATION } from '~services/GraphQL/Mutations'

export class List extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    history: PropTypes.shape({
      push: PropTypes.func.isRequired,
    }).isRequired,
    match: PropTypes.shape({
      params: PropTypes.shape({
        projectId: PropTypes.string,
      }).isRequired,
    }).isRequired,
  }

  emptyFormValues = {
    name: '',
    url: '',
    id: null,
  }

  state = {
    isDeleteModalOpen: false,
    selectedRepository: '',
  }

  handleCreate = () => {
    const { history, match } = this.props
    history.push(`${match.url}/create`)
  }

  handleEdit = ({ id }) => {
    const { history, match } = this.props
    history.push(`${match.url}/${id}`)
  }

  handleDelete = repository => {
    this.setState({ isDeleteModalOpen: true, selectedRepository: repository })
  }

  handleDeleteSubmit = async ({ delMutation }) => {
    const { selectedRepository } = this.state
    await delMutation({ variables: { id: selectedRepository.id } })
    this.handleCloseDeleteModal()
  }

  handleCloseDeleteModal = () => {
    this.setState({ isDeleteModalOpen: false })
  }

  render() {
    const { classes, match } = this.props
    const { projectId } = match.params
    const { isDeleteModalOpen, selectedRepository } = this.state

    return (
      <div className={classes.root}>
        <Typography variant="body2">
          Here you see results of all tests performed in project
        </Typography>
        <Mutation
          mutation={DELETE_REPOSITORY_MUTATION}
          refetchQueries={[
            { query: GET_TEST_SOURCES_QUERY, variables: { projectId } },
          ]}
        >
          {(delMutation, { data }) =>
            isDeleteModalOpen && (
              <DeleteModal
                onClose={this.handleCloseDeleteModal}
                onSubmit={() => this.handleDeleteSubmit({ delMutation })}
                type="repository"
                name={selectedRepository.name}
              />
            )
          }
        </Mutation>
        <div className={classes.btnContainer}>
          <AddButton onClick={this.handleCreate} />
        </div>
        <TestSourcesList
          projectId={projectId}
          onDelete={this.handleDelete}
          onEdit={this.handleEdit}
          showPagination
        />
      </div>
    )
  }
}

export default withStyles(styles)(List)
