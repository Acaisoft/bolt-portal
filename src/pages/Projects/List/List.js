import React, { useCallback } from 'react'
import PropTypes from 'prop-types'

import { ProjectsList } from './components'

import { getUrl } from '~utils/router'
import routes from '~config/routes'

export function List({ history }) {
  const getProjectDetailsUrl = useCallback(
    ({ id }) => {
      return getUrl(routes.projects.details, { projectId: id })
    },
    [history]
  )

  return <ProjectsList getProjectDetailsUrl={getProjectDetailsUrl} />
}
List.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
  match: PropTypes.shape({
    url: PropTypes.string.isRequired,
  }).isRequired,
}

export default List
