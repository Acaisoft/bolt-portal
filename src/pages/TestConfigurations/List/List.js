import React, { useCallback } from 'react'
import PropTypes from 'prop-types'


import { getUrl } from '~utils/router'
import routes from '~config/routes'
import { TestConfigurationsList } from './components'
import useStyles from './List.styles'
import { useNotification } from '~hooks'

export function List({ history, match }) {
  const { projectId } = match.params
  const classes = useStyles()

  const { handleCreate, handleDetails } = useHandlers(history, match)

  return (
    <div className={classes.root}>
      <TestConfigurationsList
        projectId={projectId}
        onCreate={handleCreate}
        onDetails={handleDetails}
      />
    </div>
  )
}
List.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      projectId: PropTypes.string,
    }).isRequired,
  }).isRequired,
}

function useHandlers(history, match) {
  const redirectToPage = useCallback(
    (path, params = {}) => {
      history.push(getUrl(path, { ...match.params, ...params }))
    },
    [history, match]
  )

  const handleCreate = useCallback(() => {
    redirectToPage(routes.projects.configurations.create)
  }, [])

  const handleDetails = useCallback(configuration => {
    redirectToPage(routes.projects.configurations.details, {
      configurationId: configuration.id,
    })
  }, [])

  const notify = useNotification()

  const handleRun = useCallback(({ configuration, error }) => {
    if (error) {
      notify.error(error)
    } else {
      notify.success(`Scenario '${configuration.name}' was successfully started.`)
    }
  }, [])

  return { handleCreate, handleDetails, handleRun }
}

export default List
