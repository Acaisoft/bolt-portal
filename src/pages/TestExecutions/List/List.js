/**
 * Copyright (c) 2022 Acaisoft
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of
 * this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to
 * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
 * the Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
 * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
 * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
 * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
 * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import React, { useCallback } from 'react'
import { useParams } from 'react-router-dom'
import { Typography } from '@material-ui/core'

import routes from 'config/routes'
import { getUrl } from 'utils/router'

import { TestExecutionsList } from './components'

import useStyles from './List.styles'

export function List() {
  const params = useParams()
  const { projectId } = params

  const classes = useStyles()

  const getExecutionDetailsUrl = useCallback(
    execution => {
      return getUrl(routes.projects.configurations.executions.details, {
        ...params,
        configurationId: execution.configuration.id,
        executionId: execution.id,
      })
    },
    [params]
  )

  return (
    <div className={classes.root}>
      <Typography variant="body2">
        Here you see results of all tests performed in all of your projects
      </Typography>
      <div className={classes.tableContainer}>
        <TestExecutionsList
          projectId={projectId}
          getExecutionDetailsUrl={getExecutionDetailsUrl}
        />
      </div>
    </div>
  )
}

export default List
