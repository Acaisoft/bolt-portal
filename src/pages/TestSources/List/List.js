import React, { useCallback } from 'react'
import { useNavigate, useParams } from 'react-router-dom'

import { TestSourcesList } from './components'

import routes from 'config/routes'
import { getUrl } from 'utils/router'

import useStyles from './List.styles'

export function List() {
  const navigate = useNavigate()
  const params = useParams()
  const { projectId } = params
  const classes = useStyles()

  const { getCreateTestSourceUrl, getEditTestSourceUrl } = useHandlers(
    navigate,
    params
  )

  return (
    <div className={classes.root}>
      <TestSourcesList
        projectId={projectId}
        getCreateTestSourceUrl={getCreateTestSourceUrl}
        getEditTestSourceUrl={getEditTestSourceUrl}
      />
    </div>
  )
}

function useHandlers(navigate, params) {
  const getRedirectUr = useCallback(
    (path, callbackParams = {}) => {
      return getUrl(path, { ...params, ...callbackParams })
    },
    [params]
  )

  const getCreateTestSourceUrl = useCallback(() => {
    return getRedirectUr(routes.projects.sources.create)
  }, [getRedirectUr])

  const getEditTestSourceUrl = useCallback(
    ({ id }) => {
      return getRedirectUr(routes.projects.sources.edit, { sourceId: id })
    },
    [getRedirectUr]
  )

  return {
    getCreateTestSourceUrl,
    getEditTestSourceUrl,
  }
}

export default List
