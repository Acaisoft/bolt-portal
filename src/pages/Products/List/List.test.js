import React from 'react'
import { shallow } from 'enzyme'

import { List } from './List'

jest.mock('~components/DataTable')

const initList = overrides => {
  const mockProps = {
    match: {
      url: '/products',
    },
  }
  const wrapper = shallow(<List {...mockProps} {...overrides} />)
  const instance = wrapper.instance()
  return { instance, wrapper }
}

describe('page: Products/List', () => {
  describe('rendering', () => {
    it('should render without crashing', () => {
      const { wrapper } = initList()
      expect(wrapper).toBeTruthy()
    })

    it('should render DataTable with provided data', () => {
      const { wrapper } = initList()
      expect(wrapper).toMatchSnapshot()
    })

    it('should render action buttons for provided row data', () => {
      const fakeRowData = { id: 1, name: 'Fake product' }
      const { instance } = initList()
      const renderedActionButtons = shallow(instance.renderActions(fakeRowData))

      expect(renderedActionButtons).toMatchSnapshot()
    })
  })
})
