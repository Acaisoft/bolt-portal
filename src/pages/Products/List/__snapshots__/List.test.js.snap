// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page: Products/List rendering should render DataTable with provided data 1`] = `
<div>
  <div>
    Products list
  </div>
  <ButtonMock
    color="primary"
    component="LinkMock"
    to="/products/create"
    variant="contained"
  >
    Create a new product
  </ButtonMock>
  <DataTable
    data={
      Array [
        Object {
          "description": "Description of the first product",
          "id": 1,
          "name": "First product",
        },
        Object {
          "description": "Description of the second product",
          "id": 2,
          "name": "Second product",
        },
        Object {
          "description": "Description of the third product",
          "id": 3,
          "name": "Third product",
        },
      ]
    }
  >
    <Column
      key="id"
      render={[Function]}
      title="ID"
    />
    <Column
      key="name"
      render={[Function]}
      title="Name"
    />
    <Column
      key="actions"
      render={[Function]}
    />
  </DataTable>
</div>
`;

exports[`page: Products/List rendering should render action buttons for provided row data 1`] = `
<div>
  <IconButtonMock
    aria-label="Show details"
    component="LinkMock"
    to="/products/1/details"
  >
    <InfoMock />
  </IconButtonMock>
  <IconButtonMock
    aria-label="Edit product"
    component="LinkMock"
    to="/products/1/edit"
  >
    <EditMock />
  </IconButtonMock>
</div>
`;
